### PR TITLE
Feature/hocs 3547 overdue rework

### PR DESCRIPTION
**Pull request by philbarton**
This change removes the need for somu data being passsed to the frontend for contribution calculations.
Contribution dates are now calculated in casework, and added to the stage data.
For COMP the contributions stage is also now calculated in casework.

### CODE REVIEW

**Review by gardnersam — APPROVED** [review]
Tested locally, works well

File: src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionsProcessorImpl.java [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                .map(csi -> new ContributionSomuInspector(csi).getContributionDueLocalDate())
+                .sorted()
+                .findFirst();
+    }
+
+    Optional<ContributionStatus> highestContributionStatus(Set<SomuItem> contributionSomuItems) {
+        return highestContributionStatus(contributionSomuItems, LocalDate.now());
+    }
+
+    Optional<ContributionStatus> highestContributionStatus(Set<SomuItem> contributionSomuItems, LocalDate now) {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionsProcessorImpl.java:62]: We only seem to actually use this, do we need the other method?

File: src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionsProcessorImpl.java [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+import static uk.gov.digital.ho.hocs.casework.contributions.ContributionStatus.*;
+
+@Service
+@Slf4j
+public class ContributionsProcessorImpl implements ContributionsProcessor {
+
+    public static final String COMPLIANT_CASE_TYPE = "COMP";
+    private final SomuItemService somuItemService;
+
+    @Autowired
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionsProcessorImpl.java:25]: Don't need `@Autowired`, constructor with most parameters is implicitely `@Autowired` while using `@Service`.

File: src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionsProcessorImpl.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import java.time.LocalDate;
+import java.util.Comparator;
+import java.util.Optional;
+import java.util.Set;
+
+import static java.time.temporal.ChronoUnit.DAYS;
+import static uk.gov.digital.ho.hocs.casework.contributions.ContributionStatus.*;
+
+@Service
+@Slf4j
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionsProcessorImpl.java:19]: Logger doesn't appear to be used? Worth adding some debug logs?

File: src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionReceivedStages.java [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package uk.gov.digital.ho.hocs.casework.contributions;
+
+public enum ContributionReceivedStages {
+    MPAM_TRIAGE,
+    MPAM_TRIAGE_ESCALATE,
+    MPAM_DRAFT,
+    MPAM_DRAFT_ESCALATE;
+
+    public static boolean contains(String stage) {
```
Comment (LiamMacP) [inline src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionReceivedStages.java:9]: Could probably pull this out into a seperate helper class as repeated, only minor though so not blocking.

Comment (philbarton) [reply src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionReceivedStages.java]: It didn't really need two enums, so merged them together.

File: src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionsProcessorImpl.java (hunk 1 above)
Comment (philbarton) [reply src/main/java/uk/gov/digital/ho/hocs/casework/contributions/ContributionsProcessorImpl.java:61]: The other one is used in tests, to control the date.

**Pull request merged**